src/DictList.elm

```diff
@@ -938,6 +965,174 @@ fromDict dict =
 
 
 
+-- LIST EXTRA
+
```

Comment:
Just to keep things organized nicely, we could move `getAt` and `(!!)` and any other things actually inspired by `List.Extra` down here -- it will help a bit when changes are made to `List.Extra` and we want to track them. (So, I'd suggest organizing the source to match where we got things from, whereas re-arranging the documentation topically).

I actually had considered putting the `List.Extra` stuff in a separate module, but I can see now that that won't work, because we'll want to make use of the `DictList` constructors which we don't expose (so aren't accessible from another module).